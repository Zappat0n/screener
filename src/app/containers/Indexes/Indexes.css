.indexes {
  display: flex;
  justify-content: center;
}
