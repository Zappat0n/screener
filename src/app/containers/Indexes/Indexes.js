import Index from '../Index/Index';

const Indexes = () => {
  return (
    <>
      <Index ticker="^GSPC"/>
      <Index ticker="^NDX"/>
      <Index ticker="^STOXX50E"/>
    </>
  );
}

export default Indexes;
