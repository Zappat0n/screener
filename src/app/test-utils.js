import React from 'react';
import { createStore, combineReducers } from 'redux';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import indexReducer from './reducers/indexSlice';
import stockReducer from './reducers/stockSlice';
import profileReducer from './reducers/profileSlice';
import filterReducer from './reducers/filterSlice';

const rootReducer = combineReducers({
  indexes: indexReducer, stocks: stockReducer, profile: profileReducer, filter: filterReducer,
});

const initialState = {
  stocks: {
    AAPL: {
      price: 100,
      companyName: 'Apple Inc',
      changes: 1,
      sector: 'Technology',
    },
  },
  profile: {
    data: {
      image: '',
      companyName: 'Microsoft',
      price: 100,
      changes: 1,
      mktCap: 1000000,
      range: '0-100',
      country: 'US',
    },
  },
};

const renderComponent = (component) => render(
  <Provider store={createStore(rootReducer, initialState)}>
    <BrowserRouter>
      {component}
    </BrowserRouter>
  </Provider>,
);

export * from '@testing-library/react';
export { renderComponent };
