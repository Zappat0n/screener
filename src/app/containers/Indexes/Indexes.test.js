import React from 'react';
import '@testing-library/jest-dom';
import { screen, fireEvent } from '@testing-library/react';
import Indexes from './Indexes';
import { renderComponent, rendererForComponent } from '../../test-utils';

test('Renders Indexes correctly', () => {
  renderComponent(
    <Indexes />,
  );
  expect(screen.getByText('S&P 500')).toBeInTheDocument();
  expect(screen.getByText('NASDAQ 100')).toBeInTheDocument();
  expect(screen.getByText('ESTX 50 PR.EUR')).toBeInTheDocument();
});

test('Indexes Match SnapShot', () => {
  const tree = rendererForComponent(
    <Indexes />,
  );
  expect(tree).toMatchSnapshot();
});

/*
test('Display Stocks', () => {
  renderComponent(
    <Indexes />,
  );

  fireEvent.click(screen.getByText('S&P 500'), { button: 1 });
  expect(screen.findByText('Apple')).toBeInTheDocument();
});
*/
