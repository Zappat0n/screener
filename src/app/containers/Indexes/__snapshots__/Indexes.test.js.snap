// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indexes Match SnapShot 1`] = `
<div
  className="indexes"
>
  <a
    className="link"
    href="/index/GSPC"
    onClick={[Function]}
  >
    <div
      className="index"
    >
      <button
        type="button"
      >
        S&P 500
        <div
          className="colorBlue"
        >
          4000
           
          2
          %
        </div>
      </button>
    </div>
  </a>
  <a
    className="link"
    href="/index/NDX"
    onClick={[Function]}
  >
    <div
      className="index"
    >
      <button
        type="button"
      >
        NASDAQ 100
        <div
          className="colorBlue"
        >
          1000
           
          %
        </div>
      </button>
    </div>
  </a>
  <a
    className="link"
    href="/index/STOXX50E"
    onClick={[Function]}
  >
    <div
      className="index"
    >
      <button
        type="button"
      >
        ESTX 50 PR.EUR
        <div
          className="colorBlue"
        >
          3000
           
          %
        </div>
      </button>
    </div>
  </a>
</div>
`;
